qiskit/transpiler/passes/layout/disjoint_utils.py

```diff
@@ -164,7 +164,11 @@ def require_layout_isolated_to_component(
                 component_index = i
                 break
         if dag.find_bit(inst.qargs[1]).index not in component_sets[component_index]:
-            raise TranspilerError("Chosen layout is not valid for the target disjoint connectivity")
+            raise TranspilerError(
+                f"The qubit {inst.qargs[1].register.name}[{inst.qargs[1].index}] is "
```

Comment:
Actually I missed this before but isn't `bit.register` deprecated? Should we just use `inst.qargs[1]` here?

Reply:
Not only deprecated, but it also might be `None` which would raise an`AttributeError` here

Reply:
oopps! I see it raising in with, for example `test.python.compiler.test_transpiler.TestTranspileMultiChipTarget.test_triple_circuit_invalid_layout_4_lookahead`. Changing. How come CI is passing?

Reply:
We have a warning skip in place because there are still a couple of places in Qiskit that use that functionality that weren't fixed before the warning was added :(.  I _think_ most of them are in very old passes, though, and #10128 should have made all the necessary information in places that it wasn't previously, though. I really need to go and squash those.

Reply:
I also realize that we dont have layout information at this point, so removing reference to registers in the new text (in 025d8a05c81f43819bdab2bd142c729fe363c161)

Reply:
> I _think_ most of them are in very old passes

I made a list in https://github.com/Qiskit/qiskit/issues/10749